tester-macos/bin/app.re

```diff
@@ -6,12 +6,12 @@ open Lwt.Infix;
 module Component = {
   [@noalloc] external lwt_start: unit => unit = "ml_lwt_iter";
 
-  let otherComponent = React.reducerComponent("Other");
-  let createElement = (~children as _, ()) => {
-    ...otherComponent,
-    initialState: _ => None,
-    reducer: (x, _) => React.Update(x),
-    render: ({state, reduce}) =>
+  let component = React.component("Other");
+  let createElement = (~children as _, ()) =>
+    component(slots => {
+      let (state, setState, _slots: React.Slots.empty) =
+        React.Hooks.useState(None, slots);
```

Comment:
One of the discussions we had in reactify was around not allowing to call hooks from outside a component: https://github.com/revery-ui/reason-reactify/issues/39

Right now it seems that's still a possibility in Brisk. Have you thought about how Brisk API could allow to express that constraint? Do you think it matters now, considering the slots "harmless", self-contained implementation nature?

Reply:
I don't think this is an issue as long as we can guard consistent semantic in those cases. That said, if we decided to return hooks from render we could force to always pass "empty" slot (i.e. all slots must be used by render.)